### DISCUSSION TITLE
pod min recommended spec, max concurrent jobs

### DISCUSSION THREAD

**Discussion by moms-spaghetti** [gotenberg/gotenberg#482] [Q&A] [answered]
We are having some issues with gotenberg failing when under heavy load, being sent many jobs. 

Am running the latest version but I see in the old 6.x documentation:
On another hand, for the [HTML](https://gotenberg.dev/docs/6.x/html), [URL](https://gotenberg.dev/docs/6.x/url) and [Markdown](https://gotenberg.dev/docs/6.x/markdown) endpoints, the API does only 6 conversions in parallel. Indeed, Google Chrome misbehaves if there are too many concurrent conversions.

Is this still true the 7.x ? I am sending gotenberg HTML for conversion. 

Also I see that:
Make sure to provide enough memory and CPU requests (for instance, 512Mi and 0.2 CPU).

Are these the actual minimum specs required per pod? I can't seem to find anything additional relating to specification in the docs (or I am totally missing them). We are currently running 3 pods with memory: "50Mi" cpu: "50m".

Thanks all

**Comment by gulien** [comment, marked as answer]
> Is this still true the 7.x ? I am sending gotenberg HTML for conversion.

Nope, as a dedicated Chrome instance is created for each conversion. In other words, the more requests, the more Chrome browsers, the more CPU/memory consumptions.

> Make sure to provide enough memory and CPU requests (for instance, 512Mi and 0.2 CPU).

Minimum specifications.

To mitigate your issue, I guess you could apply one or two of those recommendations:

1. Increase CPU/memory of your pods.
2. Scale up/down according to some metrics: memory usage, CPU usage, number of running Chrome browsers (https://gotenberg.dev/docs/modules/chromium#metric).

**Reply by moms-spaghetti** [reply]
Many thanks. All makes sense. So no limit on the chrome instances and I use the metric to scale appropriately?

Is there a ballpark recommended cpu/mem figure per chrome instance?

**Reply by gulien** [reply]
Indeed, no limit (code wise, infrastructure wise is another story 😄).

No recommended CPU/memory figures AFAIK, I guess it mostly depends on what you do, what do you convert. Your best option is to monitor and adjust accordingly 🤞

**Reply by moms-spaghetti** [reply]
Many thanks man 👍 clears everything up for me.

**Reply by gulien** [reply]
Glad I’ve been able to help 😊
